src/System.Text.Primitives/System/Text/TextEncoder.cs

```diff
@@ -19,8 +19,68 @@ public enum EncodingName : byte
             //ISO8859_1 = 4,
         }
 
-        public readonly static TextEncoder Utf8 = new Utf8TextEncoding() { Encoding = EncodingName.Utf8 };
-        public readonly static TextEncoder Utf16 = new Utf16TextEncodingLE() { Encoding = EncodingName.Utf16 };
+        #endregion EncodingName enum
+
+        #region Symbol enum
+
+        public enum Symbol : ushort
+        {
+            D0 = 0,
+            D1 = 1,
+            D2 = 2,
+            D3 = 3,
+            D4 = 4,
+            D5 = 5,
+            D6 = 6,
+            D7 = 7,
+            D8 = 8,
+            D9 = 9,
+            DecimalSeparator = 10,
+            GroupSeparator = 11,
+            InfinitySign = 12,
+            MinusSign = 13,
+            PlusSign = 14,
+            NaN = 15,
+            Exponent = 16,
+        }
+
+        #endregion Symbol enum
+
+        #region Static instances
+
+        public readonly static TextEncoder Utf8 = new Utf8TextEncoder();
+        public readonly static TextEncoder Utf16 = new Utf16TextEncoderLE();
+
+        #endregion Static instances
+
+        #region Private data
+
+        private readonly EncodingName _encodingName;
+        private readonly byte[][] _symbols;                     // this could be flattened into a single array
+        private readonly ParsingTrie.Node[] _parsingTrie;       // prefix tree used for parsing
+
+        #endregion Private data
+
+        #region Properties
+
+        public EncodingName Encoding => _encodingName;
```

Comment:
This property is called Encoding, and so all of the parameters that set it should also be called "encoding", e.g. the parameter to the TextEncoder ctor.

Otherwise the PR looks good.